libs/packages/components/src/lib/editor/editor.component.ts

```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
     registerOnTouched(fn: any) {
         this._onTouched = fn;
     }
+
+    // onKeydown(event) {
+    //     if (event.key === 'Backspace' || event.key === 'Delete') {
+    //         this.length = this.length - 1;
+    //     }
+    // }
+
+    // onKeyPressed() {
+    //     this.length = this.length + 1;
+    // }
+
+    // Get the cursor current position and ma
+    getCaretCharacterOffsetWithin(element) {
+        let caretOffset = 0;
+        if (typeof window.getSelection != 'undefined') {
+            let range = window.getSelection().getRangeAt(0);
+            let preCaretRange = range.cloneRange();
+            preCaretRange.selectNodeContents(element);
+            preCaretRange.setEnd(range.endContainer, range.endOffset);
+            caretOffset = preCaretRange.toString().length;
+        } else if (
+            typeof this._document['selection'] != 'undefined' &&
+            this._document['selection'].type != 'Control'
+        ) {
+            const body: any = this._document.body as any;
+            let textRange = this._document['selection'].createRange();
+            let preCaretTextRange = body.createTextRange();
+            preCaretTextRange.moveToElementText(element);
+            preCaretTextRange.setEndPoint('EndToEnd', textRange);
+            caretOffset = preCaretTextRange.text.length;
+        }
+        return caretOffset;
+    }
+
+    onValueChange(value) {
+
+        const rawValue = value
```

Comment:
Can we move some of the logic here to a separate function and have `validateRegex` and this method share that function? It seems like this logic of doing replaceAll, and then regex check, and then inserting `mark` element is repeated in both methods